api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java

```diff
@@ -55,35 +71,192 @@ public String createGoogleStorage(final GSBucketStorage storage) {
     }
 
     public void deleteGoogleStorage(final String bucketName) {
-        final Storage client = getClient();
+        final Storage client = gcpClient.buildStorageClient(region);
         final Iterable<Blob> blobs = client.list(bucketName, Storage.BlobListOption.prefix(EMPTY_PREFIX)).iterateAll();
         blobs.forEach(blob -> blob.delete());
-        final boolean deleted = client.delete(bucketName);
+        deleteBucket(bucketName, client);
+    }
 
-        if (!deleted) {
-            throw new IllegalArgumentException(String.format("Failed to delete google data storage %s", bucketName));
+    public DataStorageListing listItems(final GSBucketStorage storage, final String path, final Boolean showVersion,
+                                        final Integer pageSize, final String marker) {
+        final List<AbstractDataStorageItem> items = new ArrayList<>();
+
+        String requestPath = Optional.ofNullable(path).orElse(EMPTY_PREFIX);
+        if (StringUtils.isNotBlank(requestPath)) {
+            requestPath = normalizeFolderPath(requestPath);
         }
+        final Storage client = gcpClient.buildStorageClient(region);
+        final String bucketName = storage.getPath();
+        checkBucketExists(bucketName, client);
+
+        final Page<Blob> blobs = client.list(bucketName,
+                Storage.BlobListOption.currentDirectory(),
+                Storage.BlobListOption.prefix(requestPath),
+                Storage.BlobListOption.pageToken(Optional.ofNullable(marker).orElse(EMPTY_PREFIX)),
+                Storage.BlobListOption.pageSize(Optional.ofNullable(pageSize).orElse(Integer.MAX_VALUE)));
+        blobs.iterateAll().forEach(blob ->
+            items.add(blob.isDirectory()
+                    ? createDataStorageFolder(blob.getName())
+                    : createDataStorageFile(blob))
+        );
+        return new DataStorageListing(blobs.getNextPageToken(), items);
+    }
+
+    public DataStorageFile createFile(final GSBucketStorage storage, final String path, final byte[] contents) {
+        final Storage client = gcpClient.buildStorageClient(region);
+
+        final String bucketName = storage.getPath();
+        checkBucketExists(bucketName, client);
+        checkBlobDoesNotExist(bucketName, path, client);
```

Comment:
Please verify that other clients (S3/AZ) do not allow to overwrite file content, it think this method is also used for update operation.